test/kubemark/resources/start-kubemark-master.sh

```diff
@@ -493,9 +496,17 @@ start-kubemaster-component "kube-controller-manager"
 start-kubemaster-component "kube-scheduler"
 start-kubemaster-component "kube-addon-manager"
 
-# Wait till apiserver is working fine.
+# Wait till apiserver is working fine or timeout.
+echo -n "Waiting for apiserver to be healthy"
+start=$(date +%s)
 until [ "$(curl 127.0.0.1:8080/healthz 2> /dev/null)" == "ok" ]; do
+	echo -n "."
 	sleep 1
+	now=$(date +%s)
+	if [ $((now - start)) -gt 180 ]; then
```

Comment:
Done. This is just waiting for apiserver to be healthy, it should've started pretty quickly above. Even restarting it 3-4 times shouldn't take too much time, so maybe 180 was enough? But 300 sgtm.